### ISSUE TITLE
Question regarding collision_pairs

### ISSUE THREAD

**Issue by Pikil1** [kjh6526/PairwiseNet#4]
Hi, 
I initially trained the neural network following the instructions in the README, but the final MSE did not converge to a satisfactory level. Then I discovered that in the **generate_dataset_multiarm_pairwise.py** file, your collision_pair seems to be designed to include all pairs except self_pair. Furthermore, in   320 enviroments , this would likely make the sampled q extremely small.Furthermore, in a 5*8*8 scenario, this would likely make the sampled q extremely small. 

I'm not sure if this is the reason why the training results became poor. Otherwise, I simply use **data_config_multiarm.yml** to generate the training data and **env_config_2arm_table.ym**l as the test set.The training parameters are also the default values.

Any guidance on best practices for dataset generation in the dual-arm setting would be greatly appreciated.

Thanks

**Comment by jihwankim-sr** [comment]
Hello,
​First of all, thank you for your interest in our research.

​Regarding the unsatisfactory experimental results, it appears that the test environment includes an obstacle (a table) that was not present in the training data. The data generated by `generate_dataset_multiarm_pairwise.py` only contains distance information between the links of the Panda robot arms. However, the `env_config_2arm_table.yml` you used for testing includes a table between the two arms, which is a type of obstacle the model has not been trained to recognize. This is likely the cause of the issue you encountered.

​1. To test in an environment without obstacles:
Please create and use `env_config_2arm.yml` instead of `env_config_2arm_table.yml`. You can simply create this by removing the obstacles section from your original configuration file.

​2. To test in an environment with obstacles (like a table):
You must use an environment that includes obstacles starting from the dataset generation stage. In this case, please use `generate_dataset_pairwise.py` instead of `generate_dataset_multiarm_pairwise.py`.

​The `generate_dataset_multiarm_pairwise.py` script you originally used is specialized for multi-arm environments without obstacles. It is designed to collect data across various setups by changing the relative positions of the two arms. This specific feature enables the zero-shot capability of PairwiseNet across different multi-arm configurations.

​In contrast, `generate_dataset_pairwise.py` is a more general script designed to generate data even in environments with obstacles. However, unlike the multi-arm specific script, it does not have the functionality to vary the positions of robots or obstacles during collection. Therefore, it may be difficult to expect generalization to different obstacle positions from a model trained with this dataset. 

​The usage of `generate_dataset_pairwise.py` is almost identical to `generate_dataset_global.py`. Please let me know if you encounter any further issues after trying these steps.

To summarize:
- ​Please change the test environment to one without obstacles, OR
- ​Regenerate the dataset using an environment that includes obstacles.

**Comment by Pikil1** [comment]
Thank you very much for your quick, detailed, and helpful response — I really appreciate it！After removing the table obstacle as you suggested, the training performance improved significantly. I ran the training for approximately 1.2M steps, and the MSE stabilized around 0.0015–0.0014, making it difficult to decrease further. From about 800k steps onward, it mainly fluctuated within this range.

I then switched the activation function from sigmoid to ReLU, which further improved the results: the MSE dropped to around 0.0008, and the accuracy reached approximately 0.986. However, after this point, the metrics again became difficult to improve, and these results still seem to be quite far from the performance reported in the paper.

**Comment by kjh6526** [comment]
I would like to check your configuration for the 2-arm test setup.

It appears you are currently using a default spacing of 1.2m between the two arms. However, the 2-arm environment used in our paper was set at 0.6m.

If you used the default settings in `generate_dataset_multiarm_pairwise.py`, the generated dataset covers a robot spacing range of 0.1m to 1.0m. Since your test setup uses a 1.2m gap, the model was forced to perform extrapolation beyond the distribution of its training data. This lack of extrapolation generalization is likely why you are seeing such performance degradation.

I recommend adjusting the distance between the arms to 0.6m to replicate our paper's results, or changing the config setting for `generate_dataset_multiarm_pairwise.py`

**Comment by Pikil1** [comment]
hi, I had already noticed this during my previous training and set the distance to 0.6. 

id: twoarm          # identifier string of the environment

name: multipanda      # name of the environment class
base_poses:
  - [0, 0, 0]
  - [0.6, 0, 0]

base_orientations:
  - 0
  - 3.1415


collision_pairs_config:
  n_configurations: 1000000
  safe_thr: 0.01

and the another file which for generating training data keep same
n_data: 3000000
n_pcd: 100

hand: False # True for Panda hand


n_radius: 5
radius_range: [0.5, 1]

n_theta: 8
theta_range: [0, 2.7489] # 2*PI * 7/8

n_phi: 8
phi_range: [0, 2.7489] # 2*PI * 7/8
